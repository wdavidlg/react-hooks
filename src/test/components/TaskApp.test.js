
const { act } = require("@testing-library/react")
const { shallow, mount } = require("enzyme")
const { default: TaskApp } = require("../../components/08-useReducer/TaskApp")
const { tasks } = require("./fixtures/tasks")


describe('Pruebas con <TaskApp />', () => {
    
    test('debe mostrarse correctamente ', () => {
        const wrapper = shallow(<TaskApp />);
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de agregar un Task', () => {

        const wrapper = mount(<TaskApp />);

        Storage.prototype.setItem = jest.fn();

        act(() => {
            wrapper.find('TaskAdd').prop('handleAdd')(tasks[0]);
            wrapper.find('TaskAdd').prop('handleAdd')(tasks[1]);
        })
        
        expect(wrapper.find('h1').text()).toBe('TaskApp( 2 )')
        expect(localStorage.setItem).toHaveBeenCalledTimes(1)
    })

    test('debe de borrar un Task', () => {
        const wrapper = shallow(<TaskApp/>);

        wrapper.find('TaskAdd').prop('handleAdd')(tasks[0])
        wrapper.find('TaskList').prop('handleDelete')(tasks[0].id)
        expect(wrapper.find('h1').text()).toBe('TaskApp( 0 )')
    })
    


    
    
    
})
