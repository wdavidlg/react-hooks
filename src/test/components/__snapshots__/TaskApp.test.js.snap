// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <TaskApp /> debe mostrarse correctamente  1`] = `
<div>
  <h1>
    TaskApp( 
    0
     )
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <TaskList
        complete={[Function]}
        handleDelete={[Function]}
        tasks={Array []}
      />
    </div>
    <div
      className="col-5 mx-auto"
    >
      <TaskAdd
        handleAdd={[Function]}
      />
    </div>
  </div>
</div>
`;
